src/code_translation/ir/nodes.py

```diff
@@ -36,6 +36,10 @@ class DataFrameNode(IRNode):
     def sql_string(self):
         pass
 
+    @abstractmethod
+    def to_cst_translation(self, sql_access_method) -> CSTTranslation:
+        pass
+
```

Comment:
this is for my intellisense. As I mentioned it would be nice if there is no dependency injection happening here but thats for some later PR :)